dove/tests/test_cmd_dove_init.rs

```diff
@@ -0,0 +1,159 @@
+mod helper;
+use crate::helper::{
+    execute_dove_at, project_start_for_init, project_remove, set_dependencies_local_move_stdlib,
+    project_build, check_dove_toml,
+};
+/// $ dove init
+#[test]
+fn test_cmd_dove_init_without_arguments() {
+    // Project name and path
+    let project_name = "demoproject_36";
+    let project_folder = project_start_for_init(project_name);
+    execute_dove_at(&["dove", "init"], &project_folder).unwrap();
+    check_dove_toml(&project_folder, project_name, None, None, None).unwrap();
+    set_dependencies_local_move_stdlib(&project_folder);
+    project_build(&project_folder).unwrap();
+    project_remove(&project_folder);
+}
+/// $ dove init -d ###
+#[test]
+fn test_cmd_dove_init_with_dialect() {
+    // Project name and path
+    let project_name = "demoproject_43";
+    for dialect in &["pont", "diem", "dfinance"] {
+        let project_folder = project_start_for_init(project_name);
+        execute_dove_at(&["dove", "init", "-d", dialect], &project_folder).unwrap();
+        check_dove_toml(&project_folder, project_name, Some(dialect), None, None).unwrap();
+        set_dependencies_local_move_stdlib(&project_folder);
+        project_build(&project_folder).unwrap();
+        project_remove(&project_folder);
+    }
+}
+/// $ dove init -d dfinance -a ###
+#[test]
+fn test_cmd_dove_init_dfinance_with_address() {
```

Comment:
Technically the three following tests can be joined into one.

Reply:
I was asked to separate this